server/util/cache/cache.go

```diff
@@ -0,0 +1,252 @@
+package cache
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+
+	gstatus "google.golang.org/grpc/status"
+)
+
+type CacheLayer string
+
+const (
+	MemoryCacheLayer CacheLayer = "memory"
```

Comment:
can you add a comment or explain how these are used? Are these standins for "redis" and "gcs", or something else?